build/args/all.gn

```diff
@@ -14,6 +14,9 @@ v8_enable_snapshot_native_code_counters = false
 # TODO(codebytere): remove when Node.js handles https://chromium-review.googlesource.com/c/v8/v8/+/3211575
 v8_scriptormodule_legacy_lifetime = true
 
+# we use this api
+v8_enable_javascript_promise_hooks = true
```

Comment:
The comment upstream says "Allow for JS promise hooks (instead of just C++)." ([ref](https://source.chromium.org/chromium/chromium/src/+/main:v8/BUILD.gn;l=341-342;drc=8e9d8e1783afe94191e294680715ab1ec59b067e))

Do we use the JS-side API? I'm vaguely worried about this being exposed to web content instead of just node.js processes.

Reply:
@clavin will also know more here but it looks from some quick investigation that that was added in [CL:3301474](https://chromium-review.googlesource.com/c/v8/v8/+/3301474). It was just a rename though so i'm not sure why that meant we had to add it as a new flag 🤔

Reply:
I remember getting a crash when this flag was off saying explicitly it was because this feature wasn't enabled. The [CL that Shelley found](https://chromium-review.googlesource.com/c/v8/v8/+/3301474) caused it:

If you look in that CL at the only changes in `src/api/api.cc` you'll see this:

```patch
 void v8::Context::SetPromiseHooks(Local<Function> init_hook,
                                   Local<Function> before_hook,
                                   Local<Function> after_hook,
                                   Local<Function> resolve_hook) {
+#ifdef V8_ENABLE_JAVASCRIPT_PROMISE_HOOKS
   i::Handle<i::Context> context = Utils::OpenHandle(this);
   i::Isolate* isolate = context->GetIsolate();

# ........................... snipped ...........................

   context->native_context().set_promise_hook_init_function(*init);
   context->native_context().set_promise_hook_before_function(*before);
   context->native_context().set_promise_hook_after_function(*after);
   context->native_context().set_promise_hook_resolve_function(*resolve);
+#else   // V8_ENABLE_JAVASCRIPT_PROMISE_HOOKS
+  Utils::ApiCheck(false, "v8::Context::SetPromiseHook",
+                  "V8 was compiled without JavaScript Promise hooks");
+#endif  // V8_ENABLE_JAVASCRIPT_PROMISE_HOOKS
 }
```

I'm not sure the whole call path, but we already call `SetPromiseHooks` somewhere and without this flag on that method just crashes due to this CL.

Before you pointed it out, I didn't notice it mention that this exposes anything to JS, and from looking at the CL I also didn't see that it does 😕 I'm not sure if that comment on the flag is off or if that just hints at their future plans?

Reply:
Cool, seems legit 👍🏻